src/tiled/preferences.cpp

```diff
@@ -118,6 +118,11 @@ Preferences::Preferences()
     }
     mSettings->setValue(QLatin1String("RunCount"), mRunCount);
     mSettings->endGroup();
+
+    // Retrieve startup settings
+    mSettings->beginGroup(QLatin1String("Startup"));
+    mOpenLastFilesOnStartup = boolValue("OpenLastFiles");
```

Comment:
This would default the setting to `false`, but it should really be enabled by default, so `boolValue("OpenLastFiles", true);`